src/typechecker/FStar.TypeChecker.Normalize.fs

```diff
@@ -2481,7 +2483,13 @@ let normalize_with_primitive_steps ps s e t =
       r
     end
 
-let normalize s e t = normalize_with_primitive_steps [] s e t
+let normalize s e t = 
+  if (Options.profile_phase (Options.ProfileNormalize)) then 
+      let (r, _) = P.profile (fun() -> normalize_with_primitive_steps [] s e t) 
+            (fun() -> Print.term_to_string t) (Options.ProfileNormalize) in
+      r
+  else
+      normalize_with_primitive_steps [] s e t
```

Comment:
Do we really need the conditional here? Shouldn't `P.profile` already test the value of `Options.ProfileNormalize`?

Reply:
It was an attempt to keep the non-profile code the same as before since the original code doesn't call BU.record_time. Anyway, I changed it to just call P.profile to let it test the condition so the code is cleaner.